packages/clay-popover/src/index.tsx

```diff
@@ -15,15 +17,41 @@ export const ALIGN_POSITIONS = [
 	'bottom-left',
 	'bottom-right',
 	'left',
+	'left-top',
+	'left-bottom',
 	'right',
+	'right-top',
+	'right-bottom',
 ] as const;
 
+const ALIGNMENTS_MAP = {
+	bottom: ['tc', 'bc'],
+	'bottom-left': ['tl', 'bl'],
+	'bottom-right': ['tr', 'br'],
+	left: ['cr', 'cl'],
+	'left-bottom': ['br', 'bl'],
+	'left-top': ['tr', 'tl'],
+	right: ['cl', 'cr'],
+	'right-bottom': ['bl', 'br'],
+	'right-top': ['tl', 'tr'],
+	top: ['bc', 'tc'],
+	'top-left': ['bl', 'tl'],
+	'top-right': ['br', 'tr'],
+} as const;
+
 interface IProps extends React.HTMLAttributes<HTMLDivElement> {
 	/**
 	 * Position in which the tooltip will be aligned to the element.
 	 */
 	alignPosition?: typeof ALIGN_POSITIONS[number];
 
+	/**
+	 * React element that the popover will align to when clicked.
+	 */
+	alignNode?: React.ReactElement & {
```

Comment:
It might be better to rename the prop to `trigger`, at first glance I thought I needed to just pass the element ref and will also create a consistency with DropDown that uses the same approach to alignment.